bin/storm.py

```diff
@@ -296,787 +239,1044 @@ def exec_storm_class(klass, jvmtype="-server", jvmopts=[], extrajars=[], args=[]
+        help="sets the number of seconds the local cluster will run for before it shuts down",
+        default=LOCAL_TTL_DEFAULT
+    )
+
+    sub_parser.add_argument(
+        "--java-debug",
+        help="lets you turn on java debugging and set the parameters passed to -agentlib:jdwp on the JDK" +
+             "e.g transport=dt_socket,address=localhost:8000 will open up a debugging server on port 8000",
+        default=None
+    )
+
+    sub_parser.set_defaults(func=local)
+
+
+def initialize_kill_subcommand(subparsers):
+    command_help = """Kills the topology with the name topology-name. Storm will
+    first deactivate the topology's spouts for the duration of
+    the topology's message timeout to allow all messages currently
+    being processed to finish processing. Storm will then shutdown
+    the workers and clean up their state.
+    """
+    sub_parser = subparsers.add_parser("kill", help=command_help)
+
+    sub_parser.add_argument("topology-name")
+
+    sub_parser.add_argument(
+        "-w", "--wait-time-secs",
+        help="""override the length of time Storm waits between deactivation and shutdown""",
+        default=None
+    )
+
+    sub_parser.set_defaults(func=kill)
+
+
+def check_positive(value):
+    ivalue = int(value)
+    if ivalue <= 0:
+        raise argparse.ArgumentTypeError("%s is not a positive integer" % value)
+    return ivalue
+
+def check_even_list(cred_list):
+    if not (len(cred_list) % 2):
+        raise argparse.ArgumentTypeError("please provide a list of cred key and value pairs")
+    return cred_list
+
+
+def initialize_upload_credentials_subcommand(subparsers):
+    command_help = """Uploads a new set of credentials to a running topology."""
+    sub_parser = subparsers.add_parser("upload-credentials", help=command_help)
+
+    sub_parser.add_argument("topology-name")
+
+    sub_parser.add_argument(
+        "-f", "--file", default=None,
+        help="""provide a properties file with credentials in it to be uploaded"""
+    )
+
+    sub_parser.add_argument(
+        "-u", "--user", default=None,
+        help="""name of the owner of the topology (security precaution)"""
+    )
+
+    sub_parser.add_argument(
+        "cred_list", nargs='*', help="List of credkeys and their values [credkey credvalue]*",
+        type=check_even_list
+    )
+
+    sub_parser.set_defaults(func=upload_credentials)
+
+
+def initialize_sql_subcommand(subparsers):
+    command_help = """Compiles the SQL statements into a Trident topology and submits it to Storm.
+    If user activates explain mode, SQL Runner analyzes each query statement
+    and shows query plan instead of submitting topology.
+    """
+
+    sub_parser = subparsers.add_parser("sql", help=command_help)
+
+    add_client_jar_options(sub_parser)
+
+    sub_parser.add_argument("sql_file", metavar="sql-file")
+    sub_parser.add_argument(
+        "topology_name", metavar="topology-name", help="should be --explain to activate explain mode"
```

Comment:
Nit: Can we use an argument group to do this instead of putting --explain into topology-name? https://docs.python.org/3/library/argparse.html#mutual-exclusion

Reply:
There are positional arguments added on L520 that don't allow mutual exclusion